toolchain/check/handle_function.cpp

```diff
@@ -212,8 +212,11 @@ static auto BuildFunctionDecl(Context& context,
       parent_scope_inst) {
     if (auto class_decl = parent_scope_inst->TryAs<SemIR::ClassDecl>()) {
       auto& class_info = context.classes().Get(class_decl->class_id);
-      CARBON_CHECK(virtual_modifier != SemIR::Function::VirtualModifier::Impl ||
-                   class_info.is_dynamic);
```

Comment:
Looks like this used to CHECK-fail on things like
```
class C {
  impl fn F();
}
```
Would be nice to add a test for that.

Reply:
Oh, sure - added in df58bcda2bf15c117b9ed76bf69a0323b9ac450c